### ISSUE TITLE
Not inferring nullability of non-nullable value compared to null

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#32703]
Should report warnings for `s.Length` and `c.F.ToString()`:
```C#
#pragma warning disable 0649
#pragma warning disable 8618

class C
{
    internal object F;
}

class Program
{
    static void F(string s, C c)
    {
        if (s == null)
            _ = s.Length;

        if (c.F != null) return;
        c.F.ToString();
    }
}
```

**Comment by gafter** [comment]
I'm not sure I agree.  These are the moral equivalent of dead code.

**Comment by jcouv** [comment]
Subsumed by https://github.com/dotnet/roslyn/issues/33526 (Pure null tests should affect both branches)

**Comment by jcouv** [comment]
Re-opening issue as it is referenced in code (those references need to be removed first).
Sorry about that.
